### DISCUSSION TITLE
What do you with such a building issue?

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [vercel/next.js#38488] [Help] [answered]
## build is failing

![green-menus-–-Deployment-Overview](https://user-images.githubusercontent.com/42802922/178144992-afce296d-007d-456e-8b49-eb9689767987.png)

My build is failing without any useful logs, locally it is fine! How do you debug such an issue?

**Comment by rob-peters-vercel** [comment, marked as answer]
This can happen when using nx if modules have been cached or restored. To fix it, there's two options:

- Add an environmental variable: NX_CACHE_DIRECTORY = /tmp/nx-cache
- Use --skip-nx-cache for every nx command

If you're still having issues after trying this, please open a Vercel support case so we can investigate further: https://vercel.com/support/request

**Comment by Ali-Hussein-dev** [comment]
I changed the node version to 16.x and till now it works fine, I will consider the other solutions in case it happens again. Thank you for your help @rob-peters-vercel .
